tools/edit.py

```
@@ -496,6 +554,43 @@ def _file_str_replace(
         snippet = self._snippet(normalized_text, m.start(), m.start() + len(new))
         return ToolResult(output=f"Replaced code in {path}\n{snippet}")
 
+    async def _cmd_str_replace_llm(
+        self, path: Path, old: Optional[str], new: Optional[str], mode: str, model: Optional[str]
+    ) -> ToolResult:
+        if old is None:
+            raise ToolError("`old_str` required for str_replace")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)